frame/evm/src/lib.rs

```diff
@@ -677,6 +678,29 @@ pub mod pallet {
 
 	#[pallet::storage]
 	pub type Suicided<T: Config> = StorageMap<_, Blake2_128Concat, H160, (), OptionQuery>;
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_: BlockNumberFor<T>) -> Weight {
+			let pov_before = get_proof_size();
+
+			let zero_account: H160 = H160::zero();
+
+			// just a dummy read to populate the pov with the intermediates nodes
+			let _ = AccountCodesMetadata::<T>::get(zero_account.clone());
+			let (_, min_gas_weight) = T::FeeCalculator::min_gas_price();
+			let (_, account_basic_weight) = Pallet::<T>::account_basic(&zero_account);
```

Comment:
Aren't we account `min_gas_weight` and `account_basic_weight` twice?

We should only account PoV here

Reply:
`ref_time` is not cached like PoV